source/developer/mobile-developer-setup.rst

```diff
@@ -14,6 +14,82 @@ If you run into any issues getting your environment set up, check the Troublesho
 Development Environment Setup
 ---------------------------------
 
+We strongly recommend the use of a computer that supports macOS to develop and build the Mattermost mobile apps as macOS is required to build the iOS app.
+
+Software Requirements
+---------------------------------
+
+Common
+~~~~~~~~~~~~~~~~~~~
+
+Irrespective of which platform you are going to develop and build, you'll need to install some prerequisite software. For macOS, we recommend using `Homebrew <https://brew.sh/>`_ as a package manager.
+
+Install `NodeJS <https://nodejs.org/en/>`_.
+ - Using Homebrew (recommended)
+ - Open a terminal and run `brew install node`
+ - Using `NVM <https://github.com/creationix/nvm>`_
+ - Download and install the package from the `NodeJS website <https://nodejs.org/en/>`_
+
+Install `Watchman <https://facebook.github.io/watchman/>`_.
+ - Using Homebrew (recommended)
+ - Open a terminal and run `brew install watchman`
+
+Assuming that you already have NodeJS installed, you can now use **npm** to install `React Native CLI Tools <http://facebook.github.io/react-native/docs/understanding-cli.html>`_ globally
+ - npm -g install react-native-cli
+
+Install yarn (used as our package manager for the mobile apps)
+ - Homebrew
+ - Open a terminal and run `brew install yarn`
+ - NPM
+ - Open a terminal and run `npm -g install yarn` 
+
+We use GitHub to host the source code so we recommend that you install `Git <https://git-scm.com/`_ in order to get the source code and contribute your changes back with `pull requests <https://help.github.com/articles/creating-a-pull-request/>`_.
+ - If you do not have git installed you can do so with Homebrew
+  - Open a terminal and run `brew install git`
+
+iOS
+~~~~~~~~~~~~~~~~~~~
+
+In order to build and run the app in iOS you'll need to install `Xcode 9 <https://idmsa.apple.com/IDMSWebAuth/login?appIdKey=891bd3417a7776362562d2197f89480a8547b108fd934911bcbea0110d07f757&path=%2Fdownload%2F&rv=1>`_. You'll also need `Cocoapods <https://cocoapods.org/>`_ to install the project’s iOS dependencies.
+
+Android
+~~~~~~~~~~~~~~~~~~~
+
+1. First you'll need to download and install Android Studio or the Android SDK command line tools.
+2. Make sure you have the following ENV VARS configured:
+ - `ANDROID_HOME` to where Android SDK is located (likely `/Users/<username>/Library/Android/sdk`)
+ - Make sure your `PATH` includes `ANDROID_HOME/tools` and `ANDROID_HOME/platform-tools`
+  
+  - On Mac, this usually requires adding the following lines to your ~/.bash_profile file:
+  
+    - `export ANDROID_HOME=/Users/<username>/Library/Android/sdk`
+    - `export PATH=$ANDROID_HOME/platform-tools:$PATH`
+    - `export PATH=$ANDROID_HOME/tools:$PATH`
+    
+   Then reload your bash configuration: 
+    `source ~/.bashprofile`
+3. In the SDK Manager using Android Studio or the SDK CLI (sdkmanager) ensure the following are installed:
+ - Android SDK Build-Tools (multiple versions)
```

Comment:
Do they need to download all these, or just one of them?

Reply:
@enahum ^